posts/basic-probabliity-in-lean.md

```diff
@@ -0,0 +1,245 @@
+The type `ℝ≥0∞` represents the nonnegative reals and infinity: the measure of a set is a nonnegative real number which in general may be infinite.
+If `P` is a probability measure, it actually takes only values up to 1.
+The tactic `simp` knows that a probability measure is finite and will use the lemmas `measure_ne_top` or `measure_lt_top` to prove that `P s ≠ ∞` or `P s < ∞`.
+
+The operations on `ℝ≥0∞` are not as nicely behaved as on `ℝ`: `ℝ≥0∞` is not a ring. For example, subtraction truncates to zero.
+If one finds that lemma `lemma_name` used to transform an equation does not apply to `ℝ≥0∞`, a good thing to try is to find a lemma named like `ENNReal.lemma_name_of_something` and use that instead (it will typically require that one variable is not infinite).
+
+For many lemmas to apply, the set `s` will need to be a measurable set. The way to express that is `MeasurableSet s`.
+
+
+## Random variables
+
+A random variable is a measurable function from a measurable space to another.
+```lean
+variable {Ω : Type*} [MeasurableSpace Ω] {X : Ω → ℝ} (hX : Measurable X)
+```
+In that code we defined a random variable `X` from the measurable space `Ω` to `ℝ` (for which the typeclass inference system finds a measurable space instance). `hX` states that `X` is measurable, which is necessary for most manipulations.
+
+If we define a measure `P` on `Ω`, we can talk about the law or distribution of `X`. It is by definition the map of the measure `P` by `X`, `P.map X`. There is no specific notation for that law.
+To say that `X` is Gaussian with mean 0 and variance 1, write `P.map X = gaussianReal 0 1`.
+
+The expectation of `X` is the integral of that function against the measure `P`, written `∫ ω, X ω ∂P`.
+The notation `P[X]` is shorthand for that expectation. In a `MeasureSpace`, we can further use the notation `𝔼[X]`.
+
+Remark: there are two types of integrals in Mathlib, Bochner integrals and Lebesgue integrals.
+The expectation notations stand for the Bochner integral, which is defined for `X : Ω → E` with `E` a normed space over `ℝ` (`[NormedAddCommGroup E] [NormedSpace ℝ E]`).
+They don't work for `Y : Ω → ℝ≥0∞` since `ℝ≥0∞` is not a normed space, but those functions can be integrated with the Lebesgue integral: `∫⁻ ω, Y ω ∂P`.
+There is no expectation notation for the Lebesgue integral.
+
+## Discrete probability
+
+In discrete probability, measurability is not an issue: every set and every function are measurable.
+The typeclass `[DiscreteMeasurableSpace Ω]` signals that every set of `Ω` is measurable and the lemma `MeasurableSet.of_discrete` provides a proof of measurability.
+To obtain measurability of a function from `Ω`, use `Measurable.of_discrete`.
+
+Any countable type with measurable singletons is a `DiscreteMeasurableSpace`. For example `ℕ` or `Fin n`.
+
+A way to define a probability measure on a discrete space `Ω` is to use the type `PMF Ω`, which stands for probability mass function.
```

Comment:
This describes a convenient way to define a probability measure on a discrete space, but it doesn't mention if `PMF  Ω` is something that should be used as an assumption in general. For instance, let's say I want to state a theorem of the form "given a probability measure P on a discrete finite space...". Should I state the theorem over `P : PMF Ω`, or instead over `[DiscreteMeasurableSpace Ω]` with `P : Measure Ω` and `[IsProbabilityMeasure Ω]`? I'm guessing the latter, but I genuinely don't know the answer!

Reply:
To be honest I did not use PMF much, so I am unsure about it. I wrote that it was "a way to define" and not "a convenient way to define" because I am not sure about that, but at the same time I felt like I had to include a word about PMF.

Several other repositories opted for not using PMF (because bundling the summability assumption can be inconvenient, as far as I can tell). For example https://github.com/google-deepmind/debate and https://github.com/leanprover/SampCert/tree/main.

My feeling is that stating the theorem for measures will integrate better with other things in Mathlib, but I lack the practice with PMF to be assertive about that.

Reply:
Indeed, this was also the choice I made in my situation! I understand the hesitation to be assertive here, but I wonder if this might be exactly the right place in which to create a convention for how these things are meant to be used

Reply:
Ok let's use that opportunity to make a choice. I added a sentence about preferring Measure over PMF in lemmas.